File: code/oldWorkspacesEclipse/workspaceMars/Poems/src/Poet.java
import java.util.*;

public class Poet {
	// pattern: verse ==> 'The' adjective noun verb adjective object
	// lists of words
	static String[] verblist = { "walks", "eats", "sleeps", "works", "studies" };
	static String[] nounlist = { "student", "professor" };
	static String[] pronounlist = { "i", "you" };
	static String[] adjlist = { "pretty", "smart", "fun" };
	static String[] objlist = { "flowers", "books" };

	static Random rand = new Random(); // a random number generator

	/*
	 * parameters: a list of words 'list' is given action: return a random word
	 * from the list of words
	 */
	static String word(String[] list) {

		// s is a random number in the range 0 to n-1
		int s = rand.nextInt(list.length);

		// print the s-th word from the list of words
		return list[s] + " ";
	}

	// action: return a random verb from verblist
	static String verb() {
		return word(verblist);
	}

	// action: return a random noun from nounlist
	static String noun() {
		return word(nounlist);
	}

	// action: return a random pronoun from pronounlist
	static String pronoun() {
		return word(pronounlist);
	}

	// action: return a random adjective from adjlist
	static String adjective() {
		return word(adjlist);
	}

	// action: return a random object from objlist
	static String object() {
		return word(objlist);
	}

	// action: print a verse
	static void printVerse() {

		// formula for a verse is
		// verse ==> 'The' adjective noun verb adjective object
		System.out.println("The " + adjective() + noun() + verb() + adjective()
				+ object());
	}

	// action: generate a stanza
	static void stanza() {
		int i = 0;

		// stanza of 4 verses
		while (i < 4) {
			printVerse();
			i++;
		}
		System.out.println();
	}

	public static void main(String args[]) {

		// generate one stanza
		stanza();
	}

}

File: code/Cpp_Stuff/poet.cpp

// pattern: verse ==> 'The' adjective noun verb adjective object

//This is the skeleton of a program to generate free style poetry


#include <iostream.h>
#include <stdlib.h>

//list of words: last item should be NULL to mark end
char* verblist[]={"walks", "eats", "sleeps", "works", "studies", NULL};
char* nounlist[] = {"student", "professor", NULL}; 
char* pronounlist[] = {"i", "you", NULL}; 
char* adjlist[] = {"pretty", "smart", "fun", NULL}; 
char* objlist[] = {"flowers", "books", NULL};

int nverbs, nnouns, npronouns, nadj, nobj;



//parameter: a word list 'list' is given 
//action: count the number of words in the list and return it 
int list_size(char* list[]) {
  int n=0;
  while(list[n] != NULL) {
	n++;
  }
  return n;
}


//action: count the number of words in all word lists and initialize
//the global variables 'nverbs', 'nnouns', 'npronouns', 'nadj', 'nobj'
void init() {
  nverbs = list_size(verblist);
  //cout << nverbs << " verbs ";
  nnouns = list_size(nounlist);
  //cout << nnouns << " nouns ";
  npronouns = list_size(pronounlist);
  //cout << npronouns << " pronouns ";
  nadj = list_size(adjlist);
  //cout << nadj << " adjectives ";
  nobj = list_size(objlist);
  //cout << nobj << " objects " << endl;
}


//parameters: a list of words 'list' and its size  'n' are given
//action: print a random word from the list of words
void word(char *list[], int n) {
  int s; 

  //rand() generates a random number in the range 0..RAND_MAX
  //thterefore s is a random number in the range 0..n
  s = rand() % n;

  //print the s-th word from the list of words
  cout << list[s] << " ";
}




//action: print a random verb
void verb() {
  word(verblist, nverbs);
}


//action: print a random noun
void noun() {
  word(nounlist, nnouns);
}


//action: print a random pronoun
void pronoun() {
  word(pronounlist, npronouns);
}


//action: print a random adjective
void adjective() {
  word(adjlist, nadj);
}


//action: print a random object
void object() {
  word(objlist, nobj);
}



//action: generate a verse
void verse() {

  cout << "The ";
  adjective();
  noun(); 
  verb(); 
  adjective();
  object();
  cout << endl;
}



//action: generate a stanza
void  stanza() {
  int i=0; 

  // stanza if 4 verses
  while (i<4) {
    verse();
    i++;
  }
  cout << endl;
}




int  main() {

  //initialize teh list of words
  init();
  
  //initialize the random number generator 
  srand(10);
  

  //generate one stanza
  stanza();


  return 0;
}
